app/controllers/spree/api/retailops/settlement_controller.rb

```diff
@@ -62,14 +62,30 @@ def mark_complete
           render text: @settlement_results.to_json
         end
 
+        # This gets called by RetailOps after a returned item has been
+        # received and accepted at the warehouse. We need to look elsehwere
+        # for hook when returned item has been received but can not be restocked.
+        #
+        # We first look for a `RetailopsReimbursementHandler` class to see if the host
+        # application has defined any custom logic. If they have, we simply
+        # pass the parameters to the defined `Reimbursement.from_retailops_add_refund`
+        # factory method and call `RetailopsReimbursementHandler#create_reimbursement`.
         def add_refund
-          ActiveRecord::Base.transaction do
-            find_order
-            assert_return params
-            @order.update!
+          if defined?(RetailopsReimbursementHandler)
+            reimbursement = RetailopsReimbursementHandler.from_retailops_add_refund(
+              params['settlement']
```

Comment:
Yeah, I found the matching outbound log on our side and it doesn't have `settlement`.  Very strange.

```
{
   "sent" : {
      "rma_id" : "20315",
      "shipping_amt" : 0,
      "refund_items" : [
         {
            "label" : "Credit (return_item 24086)",
            "amount" : "81.54",
            "refnum" : "856653",
            "quantity" : "1",
            "id" : "856653"
         }
      ],
      "subtotal_amt" : 74.98,
      "tax_amt" : 6.56,
      "return_id" : "19946",
      "options" : {},
      "product_amt" : 74.98,
      "order_refnum" : "Rxxxxxxxxx",
      "return_items" : [
         {
            "order_item_id" : "557438",
            "quantity" : "1",
            "channel_refnum" : "856653",
            "sku" : "134396"
         }
      ],
      "refund_amt" : "81.54"
   },
   "uri" : "https://huckberry.com/api/retailops/orders/add_refund.json",
   "received" : null,
   "dur" : 1,
   "received_raw" : "\"\"",
   "status_line" : "200 OK",
   "start" : 1458768510
}
```

(Only the part under `sent` corresponds to what the Spree-side code will see)